Uno.Gallery/Uno.Gallery.Shared/Views/SamplePages/ProximitySamplePage.xaml.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Linq;
+using Uno.Gallery.ViewModels;
+using Windows.ApplicationModel.Core;
+using Windows.Devices.Enumeration;
+using Windows.Devices.Sensors;
+using Windows.UI.Core;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+
+namespace Uno.Gallery.Views.Samples
+{
+	[SamplePage(SampleCategory.NonUIFeatures, "Proximity",
+		Description = "Represents a Proximit sensor. This sensor returns distance.",
+		DocumentationLink = "https://platform.uno/docs/articles/features/proximity-sensor.html",
+		DataType = typeof(ProximitySamplePageViewModel))]
+	public sealed partial class ProximitySamplePage : Page
+	{
+		public ProximitySamplePage ()
+		{
+			this.InitializeComponent();
+		}
+
+		private void ObserveReadingChangeButton_Click(object sender, RoutedEventArgs e)
+		{
+			if ((sender as Button)?.DataContext is ProximitySamplePageViewModel viewModel)
+			{
+				if (!viewModel.ObservingReadingChange)
+				{
+					viewModel.StartObserveReadingChange();
+				}
+				else
+				{
+					viewModel.StopObservingReadingChange();
+				}
+			}
+		}
+	}
+
+	public class ProximitySamplePageViewModel : ViewModelBase
+	{
+		private const string _startObservingContent = "Start observing proximity reading changes";
+		private const string _stopObservingContent = "Stop observing proximity reading changes";
+		private string _notAvailable = "Proximity is not available on this device/platform";
+
+		private ProximitySensor _proximity;
+
+		public string ButtonContent
+		{
+			get => GetProperty<string>();
+			set => SetProperty<string>(value);
+		}
+
+		public bool ObservingReadingChange
+		{
+			get => GetProperty<bool>();
+			set => SetProperty<bool>(value);
+		}
+
+		public uint? Distance
+		{
+			get => GetProperty<uint?>();
+			set => SetProperty<uint?>(value);
+		}
+
+		public bool IsDetected
+		{
+			get => GetProperty<bool>();
+			set => SetProperty<bool>(value);
+		}
+
+		public bool ProximityAvailable => _proximity != null;
+
+
+
+		public ProximitySamplePageViewModel()
+		{
+			var selector = ProximitySensor.GetDeviceSelector();
+			var devices = DeviceInformation.FindAllAsync(selector).GetAwaiter().GetResult();
+			var device = devices.FirstOrDefault();
+			if (device is not null)
+			{
+				var proximitySensor = ProximitySensor.FromId(device.Id);
```

Comment:
You will also need to make it a platform specific code. Since the `FromId` is not yet implemented on other platforms.

```suggestion
#if __ANDROID__
				proximitySensor = ProximitySensor.FromId(device.Id);
#endif
```

Reply:
Should be fixed